src/components/tiles/geometry/geometry-toolbar-registration.tsx

```diff
@@ -0,0 +1,219 @@
+import { IToolbarButtonComponentProps, registerTileToolbarButtons } from "../../toolbar/toolbar-button-manager";
+import { TileToolbarButton } from "../../toolbar/tile-toolbar-button";
+import { isPoint } from "../../../models/tiles/geometry/jxg-types";
+import { useGeometryTileContext } from "./geometry-tile-context";
+import { canSupportVertexAngle, getVertexAngle } from "../../../models/tiles/geometry/jxg-vertex-angle";
+import { UploadButton } from "../../toolbar/upload-button";
+import { useProviderTileLinking } from "../../../hooks/use-provider-tile-linking";
+import { useReadOnlyContext } from "../../document/read-only-context";
+import { useTileModelContext } from "../hooks/use-tile-model-context";
+
+import AngleLabelSvg from "../../../clue/assets/icons/geometry/angle-label.svg";
+import AddImageSvg from "../../../clue/assets/icons/geometry/add-image-icon.svg";
+import CommentSvg from "../../../assets/icons/comment/comment.svg";
+import DeleteSvg from "../../../assets/icons/delete/delete-selection-icon.svg";
+import LineLabelSvg from "../../../clue/assets/icons/geometry/line-label.svg";
+import MovableLineSvg from "../../../clue/assets/icons/geometry/movable-line.svg";
+import PointSvg from "../../../clue/assets/icons/geometry/point-icon.svg";
+import ShapesDuplicateSvg from "../../../clue/assets/icons/geometry/shapes-duplicate-icon.svg";
+import AddDataSvg from "../../../assets/icons/add-data-graph-icon.svg";
+
+const PointButton = observer(function PointButton({name}: IToolbarButtonComponentProps) {
+  const { mode, setMode } = useGeometryTileContext();
+
+  function onClick() {
+    setMode(mode === "points" ? "select" : "points");
+  }
+
+  return (
+    <TileToolbarButton
+      name={name}
+      title="Point"
+      selected={mode === "points"}
+      onClick={onClick}
+    >
+      <PointSvg/>
+    </TileToolbarButton>
+  );
+
+});
+
+const DuplicateButton = observer(function DuplicateButton({name}: IToolbarButtonComponentProps) {
+  const { content, board, handlers } = useGeometryTileContext();
+  const disableDuplicate = board && (!content?.getOneSelectedPoint(board) &&
+    !content?.getOneSelectedPolygon(board));
+
+  return (
+    <TileToolbarButton
+      name={name}
+      title="Duplicate"
+      disabled={disableDuplicate}
+      onClick={() => handlers?.handleDuplicate()}
+    >
+      <ShapesDuplicateSvg/>
+    </TileToolbarButton>
+  );
+
+});
+
+const AngleLabelButton = observer(function AngleLabelButton({name}: IToolbarButtonComponentProps) {
+  const { content, board, handlers } = useGeometryTileContext();
+  const selectedObjects = board && content?.selectedObjects(board);
+  const selectedPoints = selectedObjects?.filter(isPoint);
+  const selectedPoint = selectedPoints?.length === 1 ? selectedPoints[0] : undefined;
+  const disableVertexAngle = !(selectedPoint && canSupportVertexAngle(selectedPoint));
+  // FIXME toggling this doesn't trigger the "observer" and thus doesn't change the selected state
```

Comment:
Is this still a problem?

Reply:
Thank you for reminding me of that.  I put in a fix but it seems a bit hacky - can you review and LMK if you see a cleaner way to do it?